### ISSUE TITLE
Given better error messages for iterated entities when not using --downlevelIteration

### ISSUE THREAD

**Issue by WangZishi** [microsoft/TypeScript#16022]
**TypeScript Version:**  2.3.3

**Code**

```ts
String.fromCharCode(...new Uint8Array(0));
```

**Expected behavior:**
I would expect there is no compile error.

**Actual behavior:**
![image](https://cloud.githubusercontent.com/assets/8288105/26357956/8be81ebe-4003-11e7-88f5-6bd95d5975af.png)
![image](https://cloud.githubusercontent.com/assets/8288105/26357988/a5be3f94-4003-11e7-9ecd-9a615c0a2e9c.png)

I have to write this code:
```
String.fromCharCode(...Array.from(new Uint8Array(0)));
```

**Comment by mhegazy** [comment]
use [`--downlevelIteration`](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#new---downleveliteration).

e.g.:
```sh
tsc --downlevelIteration --lib es5,es2015.iterable a.ts
```

**Comment by DanielRosenwasser** [comment]
In cases where the value is iterable (i.e. has a `Symbol.iterator`), we should issue this error message

```
Type '{0}' can only be iterated through when using the '--downlevelIteration' flag.
```

**Comment by TomasHubelbauer** [comment]
For people who come here from search, for me, using `"target": "es6"` fixed this. Before I didn't have any `target` configuration in `tsconfig.json`. This is with TypeScript 2.8.1.

**Comment by felixhaeberle** [comment]
Changing "ES5" to "ES2015" fixed it for me.
